pkg/controllers/inventory/inventory.go

```diff
@@ -69,7 +77,9 @@ func (h *handler) OnMachineInventory(machine *v1.MachineInventory, status v1.Mac
 	}
 
 	if cluster.Status.ClusterName == "" {
-		return status, fmt.Errorf("waiting for mgmt cluster to be created for prov cluster %s/%s", machine.Namespace, machine.Spec.ClusterName)
+		msg := fmt.Sprintf("waiting for mgmt cluster to be created for prov cluster %s/%s", machine.Namespace, machine.Spec.ClusterName)
+		h.Recorder.Event(machine, corev1.EventTypeWarning, "error", msg)
```

Comment:
oh yes, I wanted to bring this up, that field is the "reason" for the event and it says that it should be set to something readable as its meant for automation of events:

```
// 'reason' is the reason this event is generated. 'reason' should be short and unique; it
	// should be in UpperCamelCase format (starting with a capital letter). "reason" will be used
	// to automate handling of events, so imagine people writing switch statements to handle them.
	// You want to make that easy.
```

I went with error as there is other instances where this is set as error, no capital. Should we set it as something like that? should we change it to Error and Warning in some constants declaration and  use the same everywhere? How does the reason being error and the EventType being a Warning works exactly?

Reply:
we should probably go to a concise set of reasons here, example "MissingData", etc . my point here is to not expose this as an error, because it will sit here waiting for a while, and that's something expected.